File: src/Python/easy/math/problem728.py
from typing import List

class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        self_dividing_numbers = []
        for i in range(left, right + 1):
            if i > 9:
                for j in range(len(list(str(i)))):
                    if int(list(str(i))[j]) == 0:
                        break
                    else:
                        if i % int(list(str(i))[j]) == 0:
                            pass
                        else:
                            break
                        if j == len(list(str(i))) - 1:
                            self_dividing_numbers.append(i)
            else :
                if i % i == 0:
                    self_dividing_numbers.append(i)
        return self_dividing_numbers

left = 1; right = 22

solution = Solution()
solution.selfDividingNumbers(left, right)

File: src/JavaScript/easy/math/problem728.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
    let self_dividing_numbers = []
    for(let i = left; i <= right; i++){
        if(i > 9){
            let splitted_nums = i.toString().split("")
            for(let j = 0; j < splitted_nums.length; j++){
                if(i % +splitted_nums[j] == 0){
                } else {
                    break
                }
                if(j == splitted_nums.length - 1){
                    self_dividing_numbers.push(i)
                }
            }
        } else {
            if(i % i == 0){
                self_dividing_numbers.push(i)
            }
        }
    }
    return self_dividing_numbers
};

// let left = 1, right = 22
let left = 47, right = 85

selfDividingNumbers(left, right)